src/sagemaker_containers/functions.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import inspect
+
+import six
+
+import sagemaker_containers as smc
+
+
+def matching_args(fn, dictionary):  # type: (function, collections.Mapping) -> dict
+    """Given a function fn and a dict dictionary, returns the function arguments that match the dict keys.
+
+    Example:
+
+        def train(channel_dirs, model_dir): pass
+
+        dictionary = {'channel_dirs': {}, 'model_dir': '/opt/ml/model', 'other_args': None}
+
+        args = smc.functions.matching_args(train, dictionary) # {'channel_dirs': {}, 'model_dir': '/opt/ml/model'}
+
+        train(**args)
+    Args:
+        fn (function): a function
+        dictionary (dict): the dictionary with the keys
+
+    Returns:
+        (dict) a dictionary with only matching arguments.
+    """
+    arg_spec = getargspec(fn)
+
+    if arg_spec.keywords:
+        return dictionary
+
+    return smc.collections.split_by_criteria(dictionary, set(arg_spec.args)).included
```

Comment:
Is set() needed here? It's not possible for a function to have duplicate argument names, is it?

Reply:
It could be a list but ```split_by_criteria``` only is receiving sets for now. I will change it to accept lists as well.